ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java

```diff
@@ -0,0 +1,468 @@
+
+      try (CloseableIterable<FileEntry> entries = CloseableIterable.withNoopClose(Iterables.concat(changed, based))) {
+        CloseableIterable<FileEntry> mayExpiredFiles = CloseableIterable.withNoopClose(
+            Lists.newArrayList(CloseableIterable.filter(entries,
+                e -> mayExpired(e, expirationConfig, partitionFreshness, expireTimestamp))));
+        CloseableIterable.filter(mayExpiredFiles, e -> willNotRetain(e, expirationConfig, partitionFreshness))
+            .forEach(e -> {
+              if (e.isChange) {
+                changeExpiredFiles.addFile(e);
+              } else {
+                baseExpiredFiles.addFile(e);
+              }
+            });
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      expireFiles(changeTable, changeSnapshot.snapshotId(), changeExpiredFiles, expireTimestamp);
+      expireFiles(baseTable, baseSnapshot.snapshotId(), baseExpiredFiles, expireTimestamp);
+    } else {
+      UnkeyedTable unkeyedTable = table.asUnkeyedTable();
+      Snapshot snapshot = unkeyedTable.currentSnapshot();
+      ExpireFiles expiredFiles = new ExpireFiles();
+      try (CloseableIterable<IcebergFileEntry> entries
+          = fileEntriesScan(unkeyedTable, partitionFilter, snapshot).entries()) {
+        CloseableIterable<IcebergFileEntry> mayExpiredFiles = CloseableIterable.withNoopClose(
+            Lists.newArrayList(CloseableIterable.filter(entries,
+                e -> mayExpired(e, expirationConfig, partitionFreshness, expireTimestamp))));
+        CloseableIterable.filter(
+                mayExpiredFiles,
+                e -> willNotRetain(e, expirationConfig, partitionFreshness))
+            .forEach(expiredFiles::addFile);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      expireFiles(unkeyedTable, snapshot.snapshotId(), expiredFiles, expireTimestamp);
+    }
+  }
+
+  private static Expression getPartitionExpression(DataExpirationConfig expirationConfig, long expireTimestamp) {
+    if (expirationConfig.expirationLevel.equals(ExpireLevel.PARTITION)) {
+      return Expressions.alwaysTrue();
+    }
+
+    Type.TypeID typeID = expirationConfig.expirationField.type().typeId();
+    if (typeID == Type.TypeID.TIMESTAMP) {
+      return Expressions.lessThanOrEqual(expirationConfig.expirationField.name(), expireTimestamp * 1000);
+    }
+    return Expressions.alwaysTrue();
+  }
+
+  private static void expireFiles(UnkeyedTable table, long snapshotId, ExpireFiles expiredFiles, long expireTimestamp) {
+    List<DataFile> dataFiles = expiredFiles.dataFiles;
+    List<DeleteFile> deleteFiles = expiredFiles.deleteFiles;
+    if (dataFiles.isEmpty() && deleteFiles.isEmpty()) {
+      return;
+    }
+    // expire data files
+    DeleteFiles delete = table.newDelete();
+    dataFiles.forEach(delete::deleteFile);
+    delete.set(SnapshotSummary.SNAPSHOT_PRODUCER, "Amoro data expiration");
```

Comment:
"Amoro data expiration" defined in `SnapshotSummary` would be better ? It's better to define it as an enumeration or a single word, rather than a value with spaces.

Reply:
Makes sense, but I'm more confused `CommitMetaProducer` already defines certain enums, should I add new enums to it? And how does this affect thrift connection? Or defining new operation constants in `SnapshotSummary`?

Reply:
Maybe we can define a static variable in `SnapshotSummry` like `DATA_EXPIRATION` ?

Reply:
+1